Developer guide updates after API modules changes (#2935)
TouK/nussknacker: docs/developers_guide/Basics.md

- `SinkFactory`
- `CustomStreamTransformer` - types of transformations depend on type of Engine
- `Service` - mainly for defining stateless enrichments
To read more see [ComponentProvider API](./Components)

> ## Execution of scenarios

Reviewer: Execution or deployment?

Author: I've replaced with "Deployment of scenarios"

> Designer use [DeploymentManager](https://github.com/TouK/nussknacker/blob/staging/ui/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/DeploymentManager.scala)

Reviewer: ```suggestion
The Designer uses [DeploymentManager](https://github.com/TouK/nussknacker/blob/staging/ui/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/DeploymentManager.scala)
```

> interface to perform actions on scenarios (deploy / cancel / etc.). All available in distribution deployment providers are located in `managers` directory and are added to designer classpath.

Reviewer: ```suggestion
interface to perform actions on scenarios (deploy / cancel / etc.). All providers that are available in distribution deployment are located in `managers` directory and are added to the designer classpath.
```

  - [AuthenticationProvider](https://github.com/TouK/nussknacker/blob/staging/security/src/main/scala/pl/touk/nussknacker/ui/security/api/AuthenticationProvider.scala)
  - [OAuth2ServiceFactory](https://github.com/TouK/nussknacker/blob/staging/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2ServiceFactory.scala)
- [CountsReporterCreator](https://github.com/TouK/nussknacker/blob/staging/ui/processReports/src/main/scala/pl/touk/nussknacker/processCounts/CountsReporter.scala)
- [NodeAdditionalInfoProvider](https://github.com/TouK/nussknacker/blob/staging/extensions-api/src/main/scala/pl/touk/nussknacker/engine/additionalInfo/NodeAdditionalInfoProvider.scala)
- [CustomProcessValidatorFactory](https://github.com/TouK/nussknacker/blob/staging/ui/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/validation/CustomProcessValidator.scala)
             
## Modules architecture and conventions

> Diagram below shows dependencies between modules. You can see on it two main groups:

Reviewer: ```suggestion
The diagram below shows dependencies between modules. You can see two main groups on it :
```

- API modules
- Utils

*API modules* contains interfaces that are needed by our core modules (on both designer and runtime side).

> In the other hand *Utils* modules contains classes build on top of API which can be used in extensions but are not mandatory. **API of *Utils* modules can be changed more often then API inside API modules**

Reviewer: ```suggestion
On the other hand *Utils* modules contain classes built on top of API which can be used in extensions but are not mandatory. **API of *Utils* modules can be changed more often than API inside API modules**
```
